rasa/core/policies/embedding_policy.py

```diff
@@ -1,694 +1,139 @@
-import copy
-import json
 import logging
-import os
-import pickle
-
-import numpy as np
-from typing import Any, List, Optional, Text, Dict, Tuple
-
-import rasa.utils.io
-from rasa.core.domain import Domain
-from rasa.core.featurizers import (
-    TrackerFeaturizer,
-    FullDialogueTrackerFeaturizer,
-    LabelTokenizerSingleStateFeaturizer,
-    MaxHistoryTrackerFeaturizer,
+from typing import Any, Dict, Optional, Text
+
+from rasa.core.constants import DEFAULT_POLICY_PRIORITY, DIALOGUE
+from rasa.core.featurizers import TrackerFeaturizer
+from rasa.core.policies.ted_policy import TEDPolicy
+from rasa.constants import DOCS_URL_POLICIES
+from rasa.utils.tensorflow.constants import (
+    LABEL,
+    HIDDEN_LAYERS_SIZES,
+    TRANSFORMER_SIZE,
+    NUM_TRANSFORMER_LAYERS,
+    NUM_HEADS,
+    MAX_SEQ_LENGTH,
+    BATCH_SIZES,
+    BATCH_STRATEGY,
+    EPOCHS,
+    RANDOM_SEED,
+    RANKING_LENGTH,
+    LOSS_TYPE,
+    SIMILARITY_TYPE,
+    NUM_NEG,
+    EVAL_NUM_EXAMPLES,
+    EVAL_NUM_EPOCHS,
+    NEG_MARGIN_SCALE,
+    REGULARIZATION_CONSTANT,
+    SCALE_LOSS,
+    USE_MAX_NEG_SIM,
+    MAX_NEG_SIM,
+    MAX_POS_SIM,
+    EMBED_DIM,
+    DROPRATE_DIALOGUE,
+    DROPRATE_LABEL,
+    DROPRATE_ATTENTION,
+    KEY_RELATIVE_ATTENTION,
+    VALUE_RELATIVE_ATTENTION,
+    MAX_RELATIVE_POSITION,
 )
-from rasa.core.policies.policy import Policy
-from rasa.core.constants import DEFAULT_POLICY_PRIORITY
-from rasa.core.trackers import DialogueStateTracker
-from rasa.utils import train_utils
-
-import tensorflow as tf
-
-# avoid warning println on contrib import - remove for tf 2
 from rasa.utils.common import raise_warning
+from rasa.utils.tensorflow.models import RasaModel
 
-tf.contrib._warning = None
 logger = logging.getLogger(__name__)
 
 
-class EmbeddingPolicy(Policy):
+class EmbeddingPolicy(TEDPolicy):
     """Transformer Embedding Dialogue Policy (TEDP)
```

Comment:
I think the docstring should describe what the class is used for / what's its intention. Repeating the name does not do that in my opinion.